javers-core/src/main/java/org/javers/core/metamodel/type/ParametrizedDehydratedType.java

```diff
@@ -19,7 +19,11 @@
         this.actualTypeArguments =
                 Lists.transform(javersType.getActualTypeArguments(), new Function<Type, Type>() {
                     public Type apply(Type input) {
-                        return typeMapper.getJaversType(input).getRawDehydratedType();
+                        JaversType javersType = typeMapper.getJaversType(input);
+                        if (!javersType.isGenericType()){
+                            return javersType.getRawDehydratedType();
+                        }
+                        return new ParametrizedDehydratedType(javersType, typeMapper);
```

Comment:
All I need to do is to manage the code repetition (with `typeMapper.getDehydratedType()`)